### ISSUE TITLE
iOS update killed GLHView and now only get initial frame

### ISSUE THREAD

**Issue by rossebv** [shogo4405/HaishinKit.swift#484]
Hello and Thank you for such an amazing library!

I have been running your library with no issues until recently when the phone was updated. Now my preview screen is not live. It shows the first frame the camera takes upon initiation or switching to the other facing camera but no longer shows live video. I have the application running on an iPad that hasnt been updated and it still runs perfectly

Is there a new variable or setting I need to include?

My video settings and stream attachment calls
```
@IBOutlet var videoView: GLHKView!

override func viewDidLoad() {
        super.viewDidLoad()
        
        //Audio setup for RTMP Streaming
        let session: AVAudioSession = AVAudioSession.sharedInstance()
        do {
            try session.setPreferredSampleRate(44_100)
            try session.setMode(AVAudioSession.Mode.default)
            try session.setActive(true)
        } catch let error  {
            print("Error Unable to initialize Audio:  \(error.localizedDescription)")
        }
        
        //RTMP Setup
        rtmpStream = RTMPStream(connection: rtmpConnection)
        rtmpStream.syncOrientation = true
        rtmpStream.captureSettings = [
            /*"fps": 30,           FPS also controlled in 15,30, or 45*/
            "sessionPreset": AVCaptureSession.Preset.hd1280x720.rawValue,
            "continuousAutofocus": false,
            "continuousExposure": false     //Use cameras focus and exposure as defaults
        ]
        rtmpStream.videoSettings = [
            "profileLevel": kVTProfileLevel_H264_Baseline_3_1,
            "maxKeyFrameIntervalDuration": 2,
            "width": 640,
            "height": 360
        ]
        rtmpStream.audioSettings = [
            "muted": audioBool,
            "sampleRate": 44_100
        ]

override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
     
        //Setup Video Preview
        //setupLivePreview()
        rtmpStream.attachAudio(AVCaptureDevice.default(for: .audio)) { error in
            print(error.description)
        }
        
        if !camConfigured {
            guard let frontCamera = AVCaptureDevice.default(.builtInWideAngleCamera, for: 
             AVMediaType.video, position: .front)
                else {
                    print("Unable to access front camera!")
                    return
            }
            rtmpStream.attachCamera(frontCamera)
            camPosition = .front
            camConfigured = true
        } else {
            guard let curCamera = AVCaptureDevice.default(.builtInWideAngleCamera, for: 
            AVMediaType.video, position: camPosition)
                else {
                    print("Unable to access front camera!")
                    return
            }
            rtmpStream.attachCamera(curCamera)
        }
        videoView.attachStream(rtmpStream)
        videoView.videoGravity = .resizeAspectFill
    }
```

Im not seeing any errors or anything just not getting a live video preview. Does not matter if I start an rtmp stream or am just looking at the preview.

**Expected behavior**
Up until recently the preview provided a live video feed from the camera. The only changes are upgrading to swift 4.2 and my phone updating. The swift 4.2 upgrade should not be an issue as the application still works on the iPad with the older iOS after reinstalling the application. The upgrade to iOS seems to be the issue.

**Desktop (please complete the following information):**
 - OS: 10.14.1
 - Browser Chrome
 - Xcode Version 10.1

**Smartphone (please complete the following information):**
 - Device: iPhone7
 -- OS: 12.1.1
- Device: iPad
-- OS: 12.0

**Comment by hastlerr** [comment]
You did it solve the problem?

**Comment by rossebv** [comment]
I have not :/ is there anything about the new update that should be killing my preview?

Thank you so much!

**Comment by hastlerr** [comment]
@rossebv you need setup avaudiosession AVAudioSession

https://stackoverflow.com/questions/51010390/avaudiosession-setcategory-swift-4-2-ios-12-play-sound-on-silent

**Comment by rossebv** [comment]
Thank you so much for your help @hastlerr! I believe I do already set up AVAudioSession here

```
    //Audio setup for RTMP Streaming
        let session: AVAudioSession = AVAudioSession.sharedInstance()
        do {
            try session.setPreferredSampleRate(44_100)
            try session.setMode(AVAudioSession.Mode.default)
            try session.setActive(true)
        } catch let error  {
            print("Error Unable to initialize Audio:  \(error.localizedDescription)")
        }
```

Has setMode been changed to setCategory? I have tried replacing setMode with the line from the stackover

```
  //Audio setup for RTMP Streaming
        let session: AVAudioSession = AVAudioSession.sharedInstance()
        do { 
              try session.setCategory(.playback, mode: .default)
              try session.setActive(true)
         } catch let error  {
            print("Error Unable to initialize Audio:  \(error.localizedDescription)")
        }
```

as well as including it in addition to the code above.

```
//Audio setup for RTMP Streaming
        let session: AVAudioSession = AVAudioSession.sharedInstance()
        do {
              try session.setPreferredSampleRate(44_100)
              try session.setMode(AVAudioSession.Mode.default)
              try session.setCategory(.playback, mode: .default)
              try session.setActive(true)
         } catch let error  {
            print("Error Unable to initialize Audio:  \(error.localizedDescription)")
        }
```

Either way I still am not getting a video stream in my GLHView. Am I missing something?

Thank you again and please excuse me if I am missing something obvious.

**Comment by rossebv** [comment]
Thank you so much @hastlerr for all your help! 

I some how missed this update to the readme which very clearly shows the syntax for exactly the situation you said I was having.

```
let session: AVAudioSession = AVAudioSession.sharedInstance()
        do {
            try session.setPreferredSampleRate(44_100)
            if #available(iOS 10.0, *) {
                try session.setCategory(.playAndRecord, mode: .default, options: [.defaultToSpeaker, .allowBluetooth])
            } else {
                session.perform(NSSelectorFromString("setCategory:withOptions:error:"), with: AVAudioSession.Category.playAndRecord, with:  [AVAudioSession.CategoryOptions.allowBluetooth])
            }
            try session.setMode(AVAudioSession.Mode.default)
            try session.setActive(true)
        } catch let error  {
            print("Error Unable to initialize Audio:  \(error.localizedDescription)")
        }
```
